db/seeds/db_populator.rb

```diff
@@ -243,6 +239,28 @@ def create_cases(casa_org, options)
         )
       end
 
+      random_case_contact_count.times do
+        create_case_contact(new_casa_case)
+      end
+
+      # guarantee at least one case contact before and after most recent past court date
+      most_recent_past_court_date = most_recent_past_court_date(new_casa_case.id)
+      if most_recent_past_court_date
+        unless case_contact_before_last_court_date?(new_casa_case.id, most_recent_past_court_date)
+          new_case_contact = create_case_contact(new_casa_case)
+          new_case_contact.occurred_at = most_recent_past_court_date - 24.hours
+          new_case_contact.save!
+          puts "new case contact id #{new_case_contact.id} before last court date"
+        end
+
+        unless case_contact_after_last_court_date?(new_casa_case.id, most_recent_past_court_date)
+          new_case_contact = create_case_contact(new_casa_case)
+          new_case_contact.occurred_at = most_recent_past_court_date + 24.hours
+          new_case_contact.save!
+          puts "new case contact id #{new_case_contact.id} after last court date"
+        end
```

Comment:
I always feel like `if` statements are easier to read then `unless` statements -- would it be possible to rewrite these like:

```
if case_contact_after_last_court_date?(...)
    new_case_contact = create_case_contact(new_casa_case)
    new_case_contact.occurred_at = most_recent_past_court_date - 24.hours
    new_case_contact.save!
    puts "new case contact id #{new_case_contact.id} before last court date"
end

if case_contact_abefore_last_court_date?(...)
    new_case_contact = create_case_contact(new_casa_case)
    new_case_contact.occurred_at = most_recent_past_court_date + 24.hours
    new_case_contact.save!
    puts "new case contact id #{new_case_contact.id} after last court date"
end
```

Reply:
Ok sounds good!

Reply:
Oh sorry, I think `unless` is fine instead of `if !`. I just wasn't sure if you could reorder the contents of the method to just use `if` statements but it sounds like you can't. I'm fine with this or if you want to put back `unless` that is okay too.